flycheck.el

```diff
@@ -3357,17 +3357,25 @@ included file."
     (while remaining-errors
       (let* ((current-error (pop remaining-errors)))
         (when (funcall sentinel current-error)
-          ;; We found an error denoting errors in the included file, so process
-          ;; all subsequent errors until an error has the current file name
-          ;; again, and find the most severe error level
+          ;; We found an error denoting errors in the included file:
+          ;; 1. process all subsequent errors until faulty include file is found
+          ;; 2. process again all subsequent errors until an error has the
+          ;;    current file name again
+          ;; 3. find the most severe error level
           (let ((current-filename (flycheck-error-filename current-error))
                 (current-level nil)
                 (faulty-include-filename (flycheck-error-filename
-                                          (car remaining-errors))))
+                                          (car remaining-errors)))
+                (faulty-include-filename-found nil)
+                (filename nil))
             (while (and remaining-errors
-                        (not (string= (flycheck-error-filename
-                                       (car remaining-errors))
-                                      current-filename)))
+                        (progn
+                          (setq filename (flycheck-error-filename (car remaining-errors)))
```

Comment:
Well, I do not really like this pattern of setting variables in conditions of `while`.  I know that it's somewhat common in Emacs Lisp, but do you think that we could refactor this to not change a variable here?  It's just easier to understand.